### ISSUE TITLE
Join Community Refactor

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#9354]
## Description
- Is `Addresses.verification_token` still needed?
  - This is not read/accessed anywhere on the client
  - This token is automatically refreshed for all of a users addresses when joining a community
- Cleanly separate/encapsulate address encoding/decoding logic for Cosmos
- Remove a check which verifies that all addresses with the same hex belong to the same user (left-over code from migration)

**Comment by timolegros** [comment]
@raykyri if we generate the canvas message server side we can/should use the `verification_token` column to store it no?

**Comment by raykyri** [comment]
Yeah, we can just start using the column again.On Oct 2, 2024, at 2:11 AM, Timothee Legros ***@***.***> wrote:﻿
@raykyri if we generate the canvas message server side we can/should use the verification_token column to store it no?

**Comment by Rotorsoft** [comment]
So, @timolegros this means we are not removing these columns from Addresses? Can you or @raykyri provide more details?

**Comment by Rotorsoft** [comment]
Moving this to the Backlog, we might want to create a bucket next with multiple tasks to move all canvas auth to the backend/API

**Comment by raykyri** [comment]
We should discuss this on Monday or Tuesday during parking lot.

Also +cc @dillchen because moving auth would likely remove the ability to create client verifiable actions which was a major request that we spent several months on early this year.

**Comment by dillchen** [comment]
Also bringing this up #9530.

Will discuss Monday

**Comment by dillchen** [comment]
related to #9028 kind of
